vllm/core/block/cpu_gpu_block_allocator.py

```diff
@@ -105,10 +96,15 @@ def __init__(self, cpu_block_allocator: BlockAllocator,
         self._swap_mapping: Dict[int, int] = {}
         self._null_block: Optional[Block] = None
 
-        self._block_ids_to_allocator: Dict[int, BlockAllocator] = {}
-        for _, allocator in self._allocators.items():
-            for block_id in allocator.all_block_ids:
-                self._block_ids_to_allocator[block_id] = allocator
+        def _block_ids_to_allocator(block_id):
+            if cpu_block_allocator.block_index_start <= block_id < cpu_block_allocator.block_index_end:  # noqa
+                return cpu_block_allocator
+            if gpu_block_allocator.block_index_start <= block_id < gpu_block_allocator.block_index_end:  # noqa
+                return gpu_block_allocator
+            raise RuntimeError(f"unrecognized {block_id=}")
+
+        self._block_ids_to_allocator: Callable[
+            [int], BlockAllocator] = _block_ids_to_allocator
```

Comment:
Move this to a method on the class, add typing

Reply:
fixed in [fe79101](https://github.com/vllm-project/vllm/pull/5968/commits/fe791010f10fd822deedaeed7bca2bdd7e78c4bf)